cfpb/cfgov-refresh
# Introducing TableBlocks! (again!)

## Overview 
(from @richaagarwal)
- I'm editing this to reflect the entirety of the work done on introducing TableBlocks since we had to revert a previous PR (https://github.com/cfpb/cfgov-refresh/pull/2295). That PR was reverted since it did not include necessary front-end work that @sebworks has introduced here (thank you!). Also shout-out to @chosak for pairing with me on some tricky back-end issues & adding other improvements here.
- This PR encompasses the front-end and back-end work needed to integrate Wagtail's TableBlocks into our use of tables. Our old `Table` organism will no longer be used, however the field cannot be removed until the migrations in this PR have run on production. Once this is merged in, I can create a clean-up task for that. 
- There is a lot here. I tried to squash some commits, however, due to some weirdness with how reverting PRs work, git really really wants to erase the work I did to reintroduce it, so I feel comfortable saying this is the best we can do.


## Testing

### Testing the editor (this is new!)
- Create a Browse page.
- Add a table to the page.
- Configure the table with the options of your choosing.
- Verify that the table has the options you chose earlier. 

### Testing that migrations did what they're supposed to 
(this was theoretically reviewed in https://github.com/cfpb/cfgov-refresh/pull/2295)
- Grab a prod dump
- Run migrations, ~~if you have the most recent prod dump you will need to run `python cfgov/manage.py migrate --fake jobmanager 0002` before `python cfgov/manage.py migrate` (unrelated to this PR, but FYI)~~
- Check the console for names of pages that were updated
- Go to one of those pages & see that the table data is still intact

## Review

- @chosak 
- @rosskarchner 
- @contolini  
- @schaferjh 

## Screenshots
<img width="926" alt="screen shot 2016-08-25 at 1 12 30 pm" src="https://cloud.githubusercontent.com/assets/1696212/17978916/329a027c-6ac6-11e6-9e29-e3cf1e23dd32.png">
<img width="1158" alt="screen shot 2016-08-25 at 11 58 27 am" src="https://cloud.githubusercontent.com/assets/1696212/17978915/32974410-6ac6-11e6-9ea0-d674bcc1d9c8.png">


## Notes

- I didn't add all the options that were available in the design manual. If we still wish to do so then I can do that.

- There are some scenarios where the options are related (first row is header vs first column is header), it might make sense to use radio buttons in those place. 

- At some point I will open a PR against Wagtail to fix issues with the Hasontable bridge.


## Checklist
